coriolis_openstack_utils/actions.py

```diff
@@ -753,3 +755,157 @@ def execute_operations(self):
+        return {"done": False, "result": None}
+
+    def equivalent_to(self, other_action):
+        if other_action.action_type == self.action_type:
+            if self.payload['source_name'] == (
+                    other_action.payload.get('source_name')):
+                src_tenant_id = self.get_source_tenant_id()
+                src_other_tenant_id = (
+                    other_action.get_source_tenant_id())
+
+                dest_tenant_id = self.get_destination_tenant_id()
+                dest_other_tenant_id = (
+                    other_action.get_destination_tenant_id())
+                if (src_tenant_id == src_other_tenant_id and
+                        dest_tenant_id == dest_other_tenant_id):
+                    return True
+        return False
+
+    def print_operations(self):
+        super(SecurityGroupCreationAction, self).print_operations()
+        secgroup_name = self.get_new_secgroup_name()
+        LOG.info(
+            "Create new destination secgroup named '%s' with source's "
+            "security rules." % (secgroup_name))
+
+    def get_new_secgroup_name(self):
+        return self.secgroup_name_format % {
+            "original": self.payload["source_name"]}
+
+    def create_secgroup_body(self, description):
+        return {"name": self.get_new_secgroup_name(),
+                "tenant_id": self.get_destination_tenant_id(),
+                "description": description}
+
+    def execute_operations(self):
+        super(SecurityGroupCreationAction, self).print_operations()
+        dest_secgroup_name = self.get_new_secgroup_name()
+        dest_tenant_id = self.get_destination_tenant_id()
+        done = self.check_already_done()
+        if done["done"]:
+            LOG.info(
+                "Security Group named '%s' already exists.",
+                dest_secgroup_name)
+            return done["result"]
+
+        LOG.info("Creating destination security group with name '%s'" %
+                 dest_secgroup_name)
+
+        description = (self.NEW_SECGROUP_DESCRIPTION %
+                       self.payload['source_name'])
+
+        body = self.create_secgroup_body(description)
+        dest_secgroup_id = security_groups.create_security_group(
+            self._destination_client, dest_tenant_id, body)
+
+        LOG.info("Adding source %s rules to destination security group '%s'" %
+                 (self.payload['source_name'], dest_secgroup_name))
+
+        src_secgroup_name = self.payload.get('source_name')
+        src_tenant_id = self.get_source_tenant_id()
+
+        source_secgroup_list = security_groups.list_security_groups(
+            self._source_client, src_tenant_id,
+            secgroup_name=src_secgroup_name)
+
+        if not source_secgroup_list:
+            raise Exception("Source security group named %s in tenant %s "
+                            "not found!" % (src_secgroup_name, src_tenant_id))
+        else:
+            source_secgroup = source_secgroup_list.pop()
```

Comment:
Here we're silently ignoring the possibility of multiple secgroups named the same (as mentioned I don't think having more that one secgroup with the same name *in the same tenant* is possible, but please double-check)
If it is possible and `len(source_secgroup_list)` can ever be greater than 1, we'll need to `raise Exception("Multiple secgroups named %s on source")` if it is.
If there can only ever be one, the code is fine, just please use `list[0]` instead of `list.pop()` if you know a list only has one element.